### PR TITLE
refactor(messages)!: Merge no-deprecated-property/parameter into no-deprecated-api

### PR DESCRIPTION
**Pull request by RandomByte**
Also sort all messages alphabetically

### CODE REVIEW

**Comment by RandomByte** [comment]
@KlattG kindly review the user-facing messages in file "**messages.ts**" (ignore any other files).

The PR shows a large set of changes because I sorted the messages alphabetically. To see which changes I made to the actual content of the messages, see https://github.com/SAP/ui5-linter/pull/324/commits/117f44d843993f95fcbadb9cc3dfb1e7efa3580e

However, there were other changes to the messages in PRs https://github.com/SAP/ui5-linter/pull/320 and https://github.com/SAP/ui5-linter/pull/316

So I thought maybe it makes sense to review all messages together in this PR, especially since we won't add many more until the UI5 linter v1.0 release.

**Review by d3xter666 — APPROVED** [review]
LGTM

File: src/linter/messages.ts [hunk 1]
```diff
@@ -176,94 +183,70 @@ export const MESSAGE_INFO = {
...
 		ruleId: RULES["no-deprecated-api"],
 
-		message: ({propertyName, className}: {propertyName: string; className: string}) =>
-			`Use of deprecated property '${propertyName}' of class '${className}'`,
-		details: ({details}: {details: string}) => details,
+		message: ({modelName}: {modelName?: string}) =>
+			`Usage of deprecated parameter 'synchronizationMode' ` +
+			`of constructor 'sap/ui/model/odata/v4/ODataModel'${modelName ? ` (model: '${modelName}')` : ""}`,
+		details: () =>
+			`As of version 1.110.0, parameter 'synchronizationMode' is obsolete and must be omitted. ` +
```
Comment (KlattG) [inline src/linter/messages.ts:194]:
```suggestion
			`As of version 1.110.0, the 'synchronizationMode' parameter is obsolete and must be omitted. ` +
```

File: src/linter/messages.ts [hunk 2]
```diff
@@ -176,94 +183,70 @@ export const MESSAGE_INFO = {
...
+			`{@link sap/ui/model/odata/v4/ODataModel#constructor See API reference}`,
```
Comment (KlattG) [inline src/linter/messages.ts:195]:
```suggestion
			`{@link sap/ui/model/odata/v4/ODataModel#constructor See the API reference}`,
```

File: src/linter/messages.ts [hunk 3]
```diff
@@ -176,94 +183,70 @@ export const MESSAGE_INFO = {
...
-	},
-
-	[MESSAGE.DUPLICATE_BOOTSTRAP_PARAM]: {
-		severity: LintMessageSeverity.Warning,
-		ruleId: RULES["no-deprecated-api"],
-
-		message: ({name, value}: {name: string; value: string}) =>
-			`Duplicate bootstrap parameter '${name}' with value '${value}'`,
+		message: ({propertyName}: {propertyName: string}) =>
+			`Use of property deprecated property '${propertyName}'.` +
```
Comment (KlattG) [inline src/linter/messages.ts:230]:
```suggestion
			`Use of deprecated property '${propertyName}'.` +
```

File: src/linter/messages.ts [hunk 4]
```diff
@@ -277,19 +260,28 @@ export const MESSAGE_INFO = {
...
+		details: ({details}: {details?: string}) => details,
+	},
+
 	[MESSAGE.NO_DIRECT_DATATYPE_ACCESS]: {
 		severity: LintMessageSeverity.Error,
 		ruleId: RULES["no-pseudo-modules"],
 
 		message: ({moduleName}: {moduleName: string}) =>
-			`Deprecated access to DataType pseudo module '${moduleName}'`,
+			`Deprecated access of DataType-pseudo module '${moduleName}'`,
```
Comment (KlattG) [inline src/linter/messages.ts:284]:
```suggestion
			`Deprecated access of DataType pseudo module '${moduleName}'`,
```

File: src/linter/messages.ts [hunk 5]
```diff
@@ -299,7 +291,7 @@ export const MESSAGE_INFO = {
 		ruleId: RULES["no-pseudo-modules"],
 
 		message: ({moduleName}: {moduleName: string}) =>
-			`Deprecated access to enum pseudo module '${moduleName}'`,
+			`Deprecated access of enum-pseudo module '${moduleName}'`,
```
Comment (KlattG) [inline src/linter/messages.ts:294]:
```suggestion
			`Deprecated access of enum pseudo module '${moduleName}'`,
```

File: src/linter/messages.ts [hunk 6]
```diff
@@ -310,52 +302,44 @@ export const MESSAGE_INFO = {
...
-
-	[MESSAGE.PARTIALLY_DEPRECATED_PARAMETERS_GET]: {
-		severity: LintMessageSeverity.Error,
-		ruleId: RULES["no-deprecated-parameter"],
-
-		message: () =>
-			`Usage of deprecated variant of 'sap/ui/core/theming/Parameters.get'`,
-		details: () => `{@link sap.ui.core.theming.Parameters#sap.ui.core.theming.Parameters.get Parameters.get}`,
+		details: () =>
+			`Do not use global variables to access UI5 modules or API. ` +
```
Comment (KlattG) [inline src/linter/messages.ts:306]:
```suggestion
			`Do not use global variables to access UI5 modules or APIs. ` +
```

File: src/linter/messages.ts [hunk 7]
```diff
@@ -310,52 +302,44 @@ export const MESSAGE_INFO = {
...
-		ruleId: RULES["no-deprecated-parameter"],
+		ruleId: RULES["no-deprecated-api"],
 
 		message: () =>
-			`Usage of deprecated parameter 'batchGroupId' in 'sap/ui/model/odata/v2/ODataModel#createEntry'`,
-		details: () => `Use parameter 'groupId' instead. ` +
-			`{@link sap.ui.model.odata.v2.ODataModel#createEntry See API reference}`,
+			`Usage of deprecated value for parameter 'oConfig.async' of constructor 'sap/ui/core/Router'`,
+		details: () =>
+			`Parameter 'oConfig.async' must be set to true. ` +
```
Comment (KlattG) [inline src/linter/messages.ts:326]:
```suggestion
			`The 'oConfig.async' parameter must be set to true. ` +
```

File: src/linter/messages.ts [hunk 8]
```diff
@@ -310,52 +302,44 @@ export const MESSAGE_INFO = {
...
+			`{@link sap/ui/core/routing/Router#constructor See API reference}`,
```
Comment (KlattG) [inline src/linter/messages.ts:327]:
```suggestion
			`{@link sap/ui/core/routing/Router#constructor See the API reference}`,
```

File: src/linter/messages.ts [hunk 9]
```diff
@@ -310,52 +302,44 @@ export const MESSAGE_INFO = {
...
-		ruleId: RULES["no-deprecated-parameter"],
+		ruleId: RULES["no-deprecated-api"],
 
 		message: () =>
-			`Usage of deprecated value for parameter 'properties' in 'sap/ui/model/odata/v2/ODataModel#createEntry'`,
-		details: () =>
-			`Passing a list of property names is deprecated. Pass the initial values as an object instead. ` +
-			`{@link sap.ui.model.odata.v2.ODataModel#createEntry See API reference}`,
+			`Usage of deprecated value for parameter 'async' of 'sap/ui/core/Component#createComponent'`,
+		details: () => `Property 'async' must be either omitted or set to true. ` +
```
Comment (KlattG) [inline src/linter/messages.ts:336]:
```suggestion
		details: () => `The 'async' property must be either omitted or set to true. ` +
```

File: src/linter/messages.ts [hunk 10]
```diff
@@ -310,52 +302,44 @@ export const MESSAGE_INFO = {
...
+			`{@link sap.ui.core.Component#createComponent See API reference}`,
```
Comment (KlattG) [inline src/linter/messages.ts:337]:
```suggestion
			`{@link sap.ui.core.Component#createComponent See the API reference}`,
```

File: src/linter/messages.ts [hunk 11]
```diff
@@ -375,53 +359,69 @@ export const MESSAGE_INFO = {
...
+	[MESSAGE.PARTIALLY_DEPRECATED_ODATA_MODEL_V2_CREATE_ENTRY]: {
 		severity: LintMessageSeverity.Error,
-		ruleId: RULES["no-deprecated-parameter"],
+		ruleId: RULES["no-deprecated-api"],
 
 		message: () =>
-			`Usage of deprecated value for parameter 'oConfig.async' of constructor 'sap/ui/core/Router'`,
+			`Usage of deprecated parameter 'batchGroupId' in 'sap/ui/model/odata/v2/ODataModel#createEntry'`,
+		details: () => `Use parameter 'groupId' instead. ` +
+			`{@link sap.ui.model.odata.v2.ODataModel#createEntry See API reference}`,
```
Comment (KlattG) [inline src/linter/messages.ts:369]:
```suggestion
		details: () => `Use the 'groupId' parameter instead. ` +
			`{@link sap.ui.model.odata.v2.ODataModel#createEntry See the API reference}`,
```

File: src/linter/messages.ts [hunk 12]
```diff
@@ -375,53 +359,69 @@ export const MESSAGE_INFO = {
...
+		severity: LintMessageSeverity.Error,
+		ruleId: RULES["no-deprecated-api"],
+
+		message: () =>
+			`Usage of deprecated value for parameter 'properties' in 'sap/ui/model/odata/v2/ODataModel#createEntry'`,
 		details: () =>
-			`Parameter 'oConfig.async' must be set to true. ` +
-			`{@link sap/ui/core/routing/Router#constructor See API reference}`,
+			`Passing a list of property names is deprecated. Pass the initial values as an object instead. ` +
+			`{@link sap.ui.model.odata.v2.ODataModel#createEntry See API reference}`,
```
Comment (KlattG) [inline src/linter/messages.ts:380]:
```suggestion
			`{@link sap.ui.model.odata.v2.ODataModel#createEntry See the API reference}`,
```

File: src/linter/messages.ts [hunk 13]
```diff
@@ -375,53 +359,69 @@ export const MESSAGE_INFO = {
...
-		severity: LintMessageSeverity.Error,
-		ruleId: RULES["parsing-error"],
-		fatal: true,
+	[MESSAGE.SPELLING_BOOTSTRAP_PARAM]: {
+		severity: LintMessageSeverity.Warning,
+		ruleId: RULES["no-deprecated-api"],
 
-		message: ({message}: {message: string}) => message,
+		message: ({oldName, newName}: {oldName: string; newName: string}) =>
+			`Outdated spelling of bootstrap parameter: '${oldName}' should be written as '${newName}'`,
```
Comment (KlattG) [inline src/linter/messages.ts:415]:
```suggestion
			`Outdated spelling of bootstrap parameter: '${oldName}'; should be written as '${newName}'`,
```

**Review by KlattG — CHANGES_REQUESTED** [review]

**Review by flovogt — APPROVED** [review]

**Pull request merged**